skrub/_datetime_encoder.py

```diff
@@ -18,62 +21,494 @@
+    -------
+    X : Pandas or Polars dataframe
+    """
+    _, px = get_df_namespace(X)
+    index = getattr(X, "index", None)
+    X_split = [X[col].to_numpy() for col in X.columns]
+    X_split = _to_datetime_2d(X_split, **kwargs)
+    X_split = {col: X_split[col_idx] for col_idx, col in enumerate(X.columns)}
+    X = pd.DataFrame(X_split, index=index)
+    # conversion is px is Polars, no-op if Pandas
+    return px.DataFrame(X)
+
+
+def _to_datetime_series(X, **kwargs):
+    """Series specialization of :func:`pandas.to_datetime`.
+
+    Parameters
+    ----------
+    X : Pandas or Polars series
+
+    Returns
+    -------
+    X : Pandas or Polars series
+    """
+    _, px = get_df_namespace(X.to_frame())
+    index = getattr(X, "index", None)
+    name = X.name
+    X = pd.to_datetime(X, **kwargs)
+    X = pd.Series(X, index=index, name=name)
+    # conversion is px is Polars, no-op if Pandas
+    return px.Series(X)
+
+
+def _to_datetime_2d_array(X, **kwargs):
+    """2d array specialization of ``_to_datetime_2d``.
+
+    Parameters
+    ----------
+    X : ndarray of shape ``(n_samples, n_features)``
+
+    Returns
+    -------
+    X_split : list of array, of shape ``n_features``
+    """
+    X_split = np.hsplit(X, X.shape[1])
+    X_split = [X_col.ravel() for X_col in X_split]
+    return _to_datetime_2d(X_split, **kwargs)
+
+
+def _to_datetime_2d(
+    X_split,
+    indices=None,
+    indice_to_format=None,
+    format=None,
+    **kwargs,
+):
+    """Convert datetime parsable columns from a 2d array or dataframe \
+        to datetime format.
+
+    The conversion is done inplace.
+
+    Parameters
+    ----------
+    X_split : list of 1d array of length n_features
+        The 2d input, chunked into a list of array. This format allows us
+        to treat each column individually and preserve their dtype, because
+        dataframe.to_numpy() casts all columns to object when at least one
+        column dtype is object.
+
+    indices : list of int, default=None
+        Indices of the parsable columns to convert.
+        If None, indices are computed using the current input X.
+
+    indice_to_format : mapping of int to str, default=None
+        Dictionary mapping column indices to their datetime format.
+        It defines the format parameter for each column when calling
+        pd.to_datetime.
+
+        If indices is None, indices_to_format is computed using the current input X.
+        If format is not None, all values of indices_to_format are format.
+
+    format : str, default=None
+        Here for compatibility with ``pandas.to_datetime`` API.
+        When format is not None, it overwrites the values in indices_to_format.
+
+    Returns
+    -------
+    X_split : list of 1d array of length n_features
+    """
+    if indices is None:
+        indices, indice_to_format = _get_datetime_column_indices(X_split)
+
+    # format overwrite indices_to_format
+    if format is not None or indice_to_format is None:
```

Comment:
it seems `indice_to_format` can never be `None` -- values in this dict may be `None` though

Reply:
Yes indeed, thanks